src/algebra/order/floor.lean

```diff
@@ -678,6 +678,19 @@ sub_nonneg_of_le $ (le_div_iff hb).1 $ floor_le _
 lemma sub_floor_div_mul_lt (a : k) (hb : 0 < b) : a - ⌊a / b⌋ * b < b :=
 sub_lt_iff_lt_add.2 $ by { rw [←one_add_mul, ←div_lt_iff hb, add_comm], exact lt_floor_add_one _ }
 
+lemma fract_div_coe_eq_div_coe_mod {m n : ℕ} :
```

Comment:
This should probably have `nat_cast` instead of `coe`; is a similar lemma true for `int`?

Reply:
OK I've just added https://github.com/leanprover-community/mathlib/pull/17321/commits/dfa0ec223612857a34f38048e0a090186bab3a77 which attempts to address this comment.

Note that the new lemma caters for `{m : ℤ} {n : ℕ}` but not `{m n : ℤ}` since there is no clean result in this place (most easily seen by consulting `int.fract_neg`).